src/components/Table/hooks/useTableSort.js

```diff
@@ -1,14 +1,33 @@
-import { camelToSnakeCase } from "neetocist";
+import { useState } from "react";
+
+import { camelToSnakeCase, isPresent, snakeToCamelCase } from "neetocist";
 import { mergeLeft } from "ramda";
 import { useHistory } from "react-router-dom";
 
 import { useQueryParams } from "hooks";
 import { buildUrl } from "utils";
 
-import { URL_SORT_ORDERS } from "../constants";
+import { URL_SORT_ORDERS, TABLE_SORT_ORDERS } from "../constants";
+
+const getSortInfoFromQueryParams = queryParams => {
+  const sortedInfo = {};
+  if (
+    isPresent(queryParams.sort_by) &&
+    isPresent(queryParams.order_by) &&
+    isPresent(TABLE_SORT_ORDERS[queryParams.order_by])
+  ) {
+    sortedInfo.field = snakeToCamelCase(queryParams.sort_by);
+    sortedInfo.order = TABLE_SORT_ORDERS[queryParams.order_by];
+  }
+
+  return sortedInfo;
+};
 
 const useTableSort = () => {
   const queryParams = useQueryParams();
+  const [sortedInfo, setSortedInfo] = useState(
+    getSortInfoFromQueryParams(queryParams)
+  );
```

Comment:
```suggestion
  const [sortedInfo, setSortedInfo] = useState(() =>
    getSortInfoFromQueryParams(queryParams)
  );

```

Ref: https://legacy.reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
@josephmathew900 We are already calling a function, I don't think wrapping in another function is necessary.